src/sources/groups.js

```diff
@@ -1,10 +1,25 @@
+/* eslint no-use-before-define: ["error", {"functions": false, "classes": false}] */
 const authz = require("../authz");
+const {Group, GROUP_RECORDS} = require("../groups");
 const {S3Source, PrivateS3Source} = require("./s3");
 
-const PRODUCTION = process.env.NODE_ENV === "production";
+
+const GroupSource = (name) => {
+  const group = new Group(name);
+
+  const BaseSource = group.isPublic
+    ? PublicGroupSource
+    : PrivateGroupSource;
+
+  // eslint-disable-next-line no-shadow
```

Comment:
[General OOP question] `GroupSource` is a class factory, what advantage does this give as opposed to making it a class itself, so that each group is an object with `name` and `bucket` set in the constructor? We previously used classes for each group, so this implementation maintains this, but I can't remember why we did that now.

Reply:
Answered a few commits later in 4eac3516194992ebc92bd7660ebce716ae4d5289 :) 

> What was the
GroupSource function briefly is now a class so that we can properly
introspect it, e.g. with instanceof.  A parameterized class wasn't
possible before removing the static _name property, so public vs.
private had to be separate still and papered over with the previous impl
as a plain function.